### PR TITLE
Close log files at cleanup

### PR DESCRIPTION
**Pull request by puiterwijk**
Signed-off-by: Patrick Uiterwijk puiterwijk@redhat.com

### CODE REVIEW

File: fedmsg_atomic_composer/composer.py [hunk 1]
```diff
@@ -59,6 +59,8 @@ def compose(self, release):
             else:
                 traceback.print_exc()
             release['result'] = 'failed'
+        finally:
+            self.cleanup_logs()
```
Comment (bowlofeggs) [inline fedmsg_atomic_composer/composer.py]: Is this meant to be `self.cleanup_logger`?

Comment (puiterwijk) [reply fedmsg_atomic_composer/composer.py]: Yes, turns out renaming a function after writing it first time doesn't always go right :D.
Fixed.

File: fedmsg_atomic_composer/composer.py [hunk 2]
```diff
@@ -71,22 +73,31 @@ def setup_logger(self, release):
...
         shutil.rmtree(release['tmp_dir'])
 
+    def cleanup_logger(self):
+        """Clean up logger to close out file handles.
+
+        After this is called, writing to self.log will get logs ending up
+        getting discarded.
+        """
+        self.log_handler.close()
+        self.log.hremoveHandler(self.log_handler)
```
Comment (bowlofeggs) [inline fedmsg_atomic_composer/composer.py]: Is this supposed to be `hremoveHandler` and not `removeHandler`?

**Review by bowlofeggs — APPROVED** [review]
This looks reasonable to me. Perhaps we could file an issue to log to syslog instead of a file as a future improvement.

**Comment by bowlofeggs** [comment]
I filed https://github.com/fedora-infra/fedmsg-atomic-composer/issues/11 about sending the logs to syslog. We can do that in the nebulous future.

**Pull request merged**